### ISSUE TITLE
yarn add --dev vs npm install --save-dev yield different results 

### ISSUE THREAD

**Issue by csvan** [yarnpkg/yarn#2115]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**
Yarn add --dev crashes, but npm install --save-dev works

**If the current behavior is a bug, please provide the steps to reproduce.**

Run `yarn add --dev gulp-zip` on a Windows 7 machine, then try `npm install --save-dev gulp-zip` on the same.

This works fine with npm, but yarn exits with:

```
yarn add v0.17.3
[1/4] Resolving packages...
[2/4] Fetching packages...
warning fsevents@1.0.15: The platform "win32" is incompatible with this module.
info "fsevents@1.0.15" is an optional dependency and failed compatibility check. Excluding it from installation.
warning fsevents@0.2.1: The platform "win32" is incompatible with this module.
info "fsevents@0.2.1" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning Incorrect peer dependency "karma@~0.12.0".
warning Incorrect peer dependency "karma@>=0.12.8".
error An unexpected error occurred: "ENOENT: no such file or directory, open 'C:\\Users\\Me\\MyProject\\node_modules\\@types\\jasmine\\index.d.ts'".
info If you think this is a bug, please open a bug report with the information provided in "C:\\Users\\Me\\MyProject\\yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/add for documentation about this command.
```

**What is the expected behavior?**
Both should install the `gulp-zip` package under `devDependencies` in the package.json file.

**Please mention your node.js, yarn and operating system version.**
Node 6.9.1, Yarn 0.17.3, Windows 7 Enterprise

**Comment by bestander** [comment]
scoped packages were fixed in 0.18.0.
@csvan, could you try again?

**Comment by csvan** [comment]
Cannot replicate with Yarn 0.18.0, closing.
